crates/red_knot_python_semantic/src/lint.rs

```diff
@@ -414,7 +414,7 @@ pub struct RuleSelection {
     /// Map with the severity for each enabled lint rule.
     ///
     /// If a rule isn't present in this map, then it should be considered disabled.
-    lints: FxHashMap<LintId, Severity>,
+    lints: FxHashMap<LintId, (Severity, LintSource)>,
```

Comment:
This is not strictly necessary for what we need today but the intent here is that we show a note in diagnostics similar to clippy, telling users why a check was reported (you enabled this rule in the configuration, you enabled it in the cli, it's enabled by default, etc)